### PR TITLE
Refactor fairly static text into ActiveRecord Models

### PR DESCRIPTION
**Pull request by pgwillia**
- Creates migrations for Status, ItemType, CirculationRule, Library and Locations. These migrations also create today's data in the db.  
- refactor to use the new ActiveRecord objects 

These tables will populate information in this table
Screenshot from 2019-09-19 12-47-54
And the UAL shield in the results
Screenshot from 2019-09-19 14-08-37

Out of scope:
- refactor SolrMarc ingest to use the same db #1760 

#1273

### CODE REVIEW

**Comment by pgwillia** [comment]
Something is up with my db:migration.  I'll take a look tomorrow, I guess still a WIP.

**Comment by pgwillia** [comment]
> Something is up with my db:migration. I'll take a look tomorrow, I guess still a WIP.

The `Locations` ActiveRecord object was loaded by the migration potentially before it is created.  I moved the initialization of the UAL_SHIELD_LIBRARY constants to that class rather than having it in a helper.

I think this is ready for review.

File: app/helpers/holdings_helper.rb [hunk 1]
```diff
@@ -17,7 +17,21 @@ def holdings(document, method)
   end
 
   def symphony_status(item)
-    SYMPHONY_STATUSES[item[:status].downcase.underscore.to_sym]
+    Status.find_by!(short_code: item[:status].downcase.underscore).name
+  rescue ActiveRecord::RecordNotFound
+    'Unknown'
```
[the remark this answers is not in the document]
Comment (pgwillia) [reply app/helpers/holdings_helper.rb]: We had decided "if an id is looked up that doesn't exist catch in rollbar and use 'Unknown' temporarily" https://github.com/ualbertalib/discovery/issues/1273#issuecomment-499577728.  Looks like I missed the Rollbar part.  I'll leave it as an exception.

**Comment by pgwillia** [comment]
> Not 100% sure that seeds are the right way to load this data (not 100% sure they're the wrong way, either, I'm just not sure what "db:seed will seed the current data (This will be important information when this gets handed to @nmacgreg )" means).
> 
> If this is data that's only being loaded when spinning up fresh databases, seeds are a good approach. If this data is going to be added to an existing production DB, doing it in a migration instead is probably going to be cleaner.

It didn't occur to me that loading data in a migration was even a thing.  It does make sense to do it that way because our intention is that the data is a capture from a point in time and will become out of date.  If it's a seed then the expectation would be that it should be kept up.

File: app/helpers/holdings_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbarnett) [reply app/helpers/holdings_helper.rb]: That makes sense, just wanted to be sure.

File: app/helpers/holdings_helper.rb [hunk 2]
```diff
@@ -27,20 +44,14 @@ def fetch_sfx_holdings(document)
...
-    SYMPHONY_LIBRARY_LOCATIONS[:neosfree]
-  ].freeze
-
   # determines whether or not ual shield should be displayed for the item
   def display_ual_shield(document)
     return false unless document['location_tesim']
-    UAL_SHIELD_LIBRARIES.each { |library| return true if document['location_tesim'].include? library }
-    false
+
+    !Location::UAL_SHIELD_LIBRARIES.detect { |library| document['location_tesim'].include? library }.nil?
```
Comment (mbarnett) [inline app/helpers/holdings_helper.rb]: Instead of `!<..>.nil?`, how about `.present?`

Comment (pgwillia) [reply app/helpers/holdings_helper.rb]: LOL.  Of course!

**Pull request merged**